crowbar_framework/app/controllers/installer/upgrades_controller.rb

```diff
@@ -153,20 +156,23 @@ def backup
       end
     end
 
-    def nodes
+    def infos
       @current_step = 8
+
       if request.post?
         respond_to do |format|
           begin
-            @service_object.disable_non_core_proposals
-            @service_object.prepare_nodes_for_os_upgrade
+            # TODO(must): Do the following actions async in background
+            # @service_object.disable_non_core_proposals
+            # @service_object.prepare_nodes_for_os_upgrade
+
```

Comment:
> This should be async because the following step will do AJAX requests to retrieve the currently still left to update nodes.

Right, but the real update just starts after "reboot" was called and by that prepare_nodes_for_os_upgrade() is already finished. Then we load the next page which does the AJAX magic, because that's where it takes really long (AutoYaST ...).